Modules/Quotas/src/Quotas.Infrastructure/Persistence/Repository/RelationshipsRepository.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Domain.Aggregates.Relationships;
+using Backbone.Modules.Quotas.Infrastructure.Persistence.Database;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backbone.Modules.Quotas.Infrastructure.Persistence.Repository;
+public class RelationshipsRepository : IRelationshipsRepository
+{
+    private readonly IQueryable<Relationship> _relationshipsReadonly;
+
+    public RelationshipsRepository(QuotasDbContext dbContext)
+    {
+        _relationshipsReadonly = dbContext.Relationships.AsNoTracking();
+    }
+
+    public async Task<uint> Count(string createdBy, DateTime createdAtFrom, DateTime createdAtTo, CancellationToken cancellationToken)
+    {
+        var relationshipsCount = await _relationshipsReadonly
+            .Where(r => r.Status == RelationshipStatus.Pending && r.From == createdBy ||
+                r.Status == RelationshipStatus.Active && (r.From == createdBy || r.To == createdBy))
+            .CountAsync(cancellationToken);
+        return (uint)relationshipsCount;
```

Comment:
I think there is too much logic for it to be untested. :(

You can use Enmeshed.UnitTestTools.TestDoubles.Fakes.FakeDbContextFactory to instantiate Sqlite DbContexts which you can use to test